lib/solarwinds_apm/support/resource_detector/aws/eks.rb

```diff
@@ -0,0 +1,147 @@
+# frozen_string_literal: true
+
+# © 2023 SolarWinds Worldwide, LLC. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at:http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+require 'net/http'
+require 'uri'
+require 'json'
+require 'socket'
+
+module SolarWindsAPM
+  module ResourceDetector
+    module EKS
+      module_function
+
+      K8S_SVC_URL = 'kubernetes.default.svc'
+      K8S_TOKEN_PATH = '/var/run/secrets/kubernetes.io/serviceaccount/token'
+      K8S_CERT_PATH = '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt'
+      AUTH_CONFIGMAP_PATH = '/api/v1/namespaces/kube-system/configmaps/aws-auth'
+      CW_CONFIGMAP_PATH = '/api/v1/namespaces/amazon-cloudwatch/configmaps/cluster-info'
+      CONTAINER_ID_LENGTH = 64
+      DEFAULT_CGROUP_PATH = '/proc/self/cgroup'
+      TIMEOUT_MS = 2000
+      UTF8_UNICODE = 'utf-8'
+
+      def detect
+        ::OpenTelemetry::SDK::Resources::Resource.create(gather_data)
+      end
+
+      def gather_data
+        raise StandardError unless File.exist?(K8S_TOKEN_PATH) && File.readable?(K8S_TOKEN_PATH)
```

Comment:
Same comment as previous could be good to have message saying skipped because not running in k8s, and also log levels should be debug.